kernel/enter.c

```diff
@@ -653,6 +662,116 @@ static void _print_boottime(void)
 /* the main thread is the only thread that is not on the heap */
 static myst_thread_t _main_thread;
 
+static void concat_env_variables(
+    const char* value,
+    myst_args_t new_envp,
+    int pos)
+{
+    size_t val_len = strlen(value);
+    size_t nv_len = strlen(new_envp.data[pos]);
+    char* tuple = malloc(nv_len + 1 + val_len);
+    strncat(tuple, new_envp.data[pos], nv_len);
+    strncat(tuple, ":", 1);
+    strncat(tuple, value, val_len);
+    if (myst_args_remove(&new_envp, pos, 1) != 0)
+    {
+        myst_eprintf("path variable not found in envp");
+    }
+    myst_args_append1(&new_envp, tuple);
```

Comment:
I think `tuple` can/should be freed after append. Can you double check?

Reply:
looking into the frees